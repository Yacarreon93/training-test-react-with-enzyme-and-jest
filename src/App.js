import React, { useState } from 'react';
import logo from './logo.svg';
import './App.css';

export const Title = ({ text }) => <div>{text}</div>;

export class Link extends React.Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>;
  }
}

class App extends React.Component {
  state = {
    on: false,
    color: 'blue',
    input: '',
    lifeCycle: '',
  };

  UNSAFE_componentWillReceiveProps() {
    this.setState({ lifeCycle: 'UNSAFE_componentWillReceiveProps' });
  }

  componentDidMount() {
    this.setState({ lifeCycle: 'componentDidMount' });
  }

  handleStrings(str) {
    return !!str;
  }

  render() {
   const {
      on,
      color,
      input,
      lifeCycle,
    } = this.state;

    return (
      <div className="App">
        <header className="App-header">
          <h1>Welcome to React</h1>
          <h2>{input}</h2>
          <img src={logo} className="App-logo" alt="logo" />
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
        <ul className="custom-list">
          <li className={color}></li>
          <li></li>
          <li></li>
        </ul>
        <Title text="Hello" />
        <p className="button-state">{on ? 'Yes!' : 'No!'}</p>
        <button onClick={() => this.setState({ on: !on })}>Click</button>
        <input type="text" onChange={e => this.setState({ input: e.currentTarget.value })} />
        <span className="lifeCycle">{lifeCycle}</span>
      </div>
    );
  }
}

export default App;
