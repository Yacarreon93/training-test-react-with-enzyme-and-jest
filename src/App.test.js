import React from 'react';
import { render } from '@testing-library/react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import App from './App';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it ('should render App', () => {
    const wrapper = shallow(<App />, { context: {}, disableLifecycleMethods: true });

    console.log(wrapper.debug());
  });
});
