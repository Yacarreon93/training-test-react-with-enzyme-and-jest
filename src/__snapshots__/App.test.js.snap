// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Welcome to React
    </h1>
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <ul
    className="custom-list"
  >
    <li />
    <li />
    <li />
  </ul>
  <Title
    text="Hello"
  />
</div>
`;

exports[`<App /> matches the snapshot 2`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Welcome to React
    </h1>
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <ul
    className="custom-list"
  >
    <li />
    <li />
    <li />
  </ul>
  <Title
    text="Hello"
  />
</div>
`;
